Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1370,12 +1373,13 @@ ExceptionOr<void> HTMLElement::showPopover()
 
     ASSERT(!isInTopLayer());
 
+    auto& originalDocument = document();
```

Comment:
```suggestion
    Ref document = this->document();
```

Please use `Ref` here. Also I'd just call it document and use it in the rest of the function (replacing the calls to `document()`).